runtime-sdk/src/crypto/signature/ed25519.rs

```diff
@@ -94,3 +115,63 @@ impl From<PublicKey> for CorePublicKey {
         pk.0
     }
 }
+
+/// A memory-backed signer for Ed25519.
+pub struct MemorySigner {
+    sk: ed25519_dalek::ExpandedSecretKey,
+}
+
+impl MemorySigner {
+    pub fn sign_digest<D>(&self, digest: D) -> Result<Signature, Error>
+    where
+        D: ed25519_dalek::Digest<OutputSize = U64>,
+    {
+        let pk = ed25519_dalek::PublicKey::from(&self.sk);
+        self.sk
+            .sign_prehashed(digest, &pk, None)
+            .map_err(|_| Error::SigningError)
+            .map(|sig| sig.to_bytes().to_vec().into())
+    }
+}
+
+impl super::Signer for MemorySigner {
+    fn new_from_seed(seed: &[u8]) -> Result<Self, Error> {
```

Comment:
this isn't really so much of `generate` as `derivePublicKey`. Maybe this should invoke `random_bytes` with `seed` as the pers str?